File: Assets/AltUnityTester/AltUnityDriver/Commands/InputActions/PressKeyAndWait.cs
public class PressKeyAndWait : AltBaseCommand
{
    Assets.AltUnityTester.AltUnityDriver.UnityStruct.KeyCode keyCode;
    float power;
    float duration;
    public PressKeyAndWait(SocketSettings socketSettings, Assets.AltUnityTester.AltUnityDriver.UnityStruct.KeyCode keyCode, float power, float duration) : base(socketSettings)
    {
        this.keyCode = keyCode;
        this.power = power;
        this.duration = duration;
    }
    public void Execute(){
        new PressKey(SocketSettings,keyCode, power, duration).Execute();
        System.Threading.Thread.Sleep((int)duration * 1000);
        string data;
        do
        {
            Socket.Client.Send(toBytes(CreateCommand("actionFinished")));
            data = Recvall();
        } while (data == "No");
        if (data.Equals("Yes"))
            return;
        HandleErrors(data);
    }
}

File: Assets/AltUnityTester-BindingsAndExamples(can_be_deleted)/Bindings~/python/altunityrunner/commands/InputActions/press_key_and_wait.py
from altunityrunner.commands.command_returning_alt_elements import CommandReturningAltElements
from altunityrunner.commands.InputActions.press_key import PressKey
import time
class PressKeyAndWait(CommandReturningAltElements):
    def __init__(self, socket,requestSeparator,requestEnd, keyName,power,duration):
        super().__init__(socket,requestSeparator,requestEnd)
        self.keyName=keyName
        self.power=power
        self.duration=duration
    
    def execute(self):
        data = PressKey(self.socket,self.requestSeparator,self.requestEnd,self.keyName,self.power,self.duration).execute()
        self.handle_errors(data)
        print('Wait for press key to finish')
        time.sleep(self.duration)
        action_in_progress = True
        while action_in_progress:
            action_finished = self.send_data(self.create_command('actionFinished'))
            self.handle_errors(action_finished)
            if action_finished is 'Yes':
                break
            elif action_finished != 'No':
                action_in_progress = False
        return self.handle_errors(data)